app/controllers/embedded_terraform_credential_controller.rb

```diff
@@ -0,0 +1,148 @@
+class EmbeddedTerraformCredentialController < ApplicationController
+  before_action :check_privileges
+  before_action :get_session_data
+
+  after_action :cleanup_action
+  after_action :set_session_data
```

Comment:
If we do a prototype, each of the controllers would probably need a before_action like workflows:

```ruby
app/controllers/workflow_controller.rb-class WorkflowController < ApplicationController
app/controllers/workflow_controller.rb:  before_action :check_prototype
```

Maybe we can generalize the check prototype mixin to avoid duplicating it.

```ruby
app/controllers/mixins/workflow_check_prototype_mixin.rb-module Mixins
app/controllers/mixins/workflow_check_prototype_mixin.rb-  module WorkflowCheckPrototypeMixin
app/controllers/mixins/workflow_check_prototype_mixin.rb-    private
app/controllers/mixins/workflow_check_prototype_mixin.rb-
app/controllers/mixins/workflow_check_prototype_mixin.rb:    def check_prototype
app/controllers/mixins/workflow_check_prototype_mixin.rb:      return if Settings.prototype.ems_workflows.enabled
app/controllers/mixins/workflow_check_prototype_mixin.rb-
app/controllers/mixins/workflow_check_prototype_mixin.rb-      log_privileges(false, "Workflows are not enabled. The user is not authorized for this task or item.")
app/controllers/mixins/workflow_check_prototype_mixin.rb-      raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.')
app/controllers/mixins/workflow_check_prototype_mixin.rb-    end
app/controllers/mixins/workflow_check_prototype_mixin.rb-  end
app/controllers/mixins/workflow_check_prototype_mixin.rb-end
```

Thoughts on generalizing this to:

```ruby
class WorkflowController < ApplicationController
  include CheckPrototypeMixin
  before_action do 
    check_prototype(:ems_workflows)
  end
...
```

```ruby
class EmbeddedTerraformCredentialController < ApplicationController
  include CheckPrototypeMixin
  before_action do 
    check_prototype(:embedded_terraform)
  end
...
```

and the mixin becomes more generic:

```ruby
module Mixins
  module CheckPrototypeMixin
    private

    def check_prototype(feature)
      return if Settings.prototype.send(feature)&.enabled

      log_privileges(false, "#{feature} are not enabled. The user is not authorized for this task or item.")
      raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.')
    end
  end
end
```

We probably need to make sure the setting can be converted to human readable "Workflows are not enabled" instead of "ems_workflows are not enabled".

Reply:
Okay I added a prototype flag to the controllers and menu item so they will only show up if Settings.prototype.ems_embedded_terraform.enabled is true